app/models/public_comment.rb

```diff
@@ -1,30 +1,32 @@
 class PublicComment < Comment
 
-  after_create :create_notifications, :send_emails
+  after_create :notify, :send_emails
 
   private
 
   # Send emails to speakers when reviewer creates a comment
   def send_emails
     if user.reviewer_for_event?(proposal.event)
-      ProposalMailer.comment_notification(proposal, self).deliver_now
+      CommentNotificationMailer.speaker_notification(proposal, self).deliver_now
+    else
+      CommentNotificationMailer.reviewer_notification(self).deliver_now
     end
   end
 
-  # Generate notifications for Comment
+  # Generate notifications for PublicComment
   # 1. If a speaker is leaving a comment,
   #    all reviewers/organizers get an in app notification
-  #      if they have reviewed or commented on the proposal.
+  #      if they have reviewed/rated or commented on the proposal.
   # 2. If a a reviewer/organizer leaves a comment,
-  #      only the speakers get an in app and email notification.
-  def create_notifications
+  #  only the speakers get an in app and email notification.
 
+  def notify
     if user.reviewer_for_event?(proposal.event)
       users = proposal.speakers.map(&:user)
-      message = "#{user.name} has commented on #{proposal.title}"
+      message = "New comment on #{proposal.title}"
     else
       users = proposal.reviewers
```

Comment:
so the one trick with reusing this list of users is that when you pass it into the mailer method, add the .with_notifications scope so that only reviewers with notifications are passed into the mailer call.